src/simulators/matrix_product_state/matrix_product_state_internal.hpp

```diff
@@ -382,6 +386,15 @@ class MPS{
   uint_t num_qubits_;
   std::vector<MPS_Tensor> q_reg_;
   std::vector<rvector_t> lambda_reg_;
+
+  // qubit_order_ stores the current ordering of the qubits,
+  // qubit_location_ stores the location of each qubit in the vector. It is derived from qubit_order_ 
+  // at the end of every swap operation for performance reasons
+  // for example: starting position qubit_order_=qubit_location_=01234
+  // ccx(0,4) -> qubit_order_=04123, qubit_location_=02341
+  reg_t qubit_order_;
```

Comment:
nit: A nice abstraction for `qubit_order_` and `qubit_location_` would be creating a simple class to encapsulate both, or a `struct` if it's very simple. This way we enforce the fact that they are tightly coupled.

Reply:
I agree. done.